api/lib/domain/models/CertificationCourse.js

```diff
@@ -151,7 +151,7 @@ class CertificationCourse {
     };
   }
 
-  get id() {
+  getId() {
```

Comment:
je ne comprends pas l'interêt du remplacement. C'est pour être plus explicite?

Reply:
(J'aime bien les getter js)

Reply:
(moi j'aime pas ^^) je vois pas l'intérêt de "simuler" l'accès à un champ via une fonction. Je peux comprendre ça en mode "Oups, j'ai besoin d'ajouter du comportement alors que j'accède à ce champs et j'ai pas encapsulé donc je trouve une combiné" (et même dans ce cas là j'ai envie de dire "bah encapsule maintenant c'est pas trop tard") mais en première intention, je vois pas...

Reply:
Autre truc que j'aime bien c'est que ça m'amène à faire `myObject.getId()` plutôt que `myObject.id`.

Dans le premier cas si l'objet n'a pas de fonction `getId()` j'obtiens un `getId is not a function` **en lieu et place de l'appel**. Dans le deuxième cas, si `id` n'existe pas j'obtiens un `undefined` qui, dans le meilleur des cas, provoque un problème 12 mètres plus loin et me laisse le soin de remonter comme un grand à la source du problème et, dans le pire des cas, ne plante pas du tout et potentiellement a des effets néfastes silencieux dont je me rends compte à retardement (Ok, avec les tests qui vont bien ça ne devrait pas se produire mais on est pas infaillibles). 

Donc utiliser `getId()` plutôt que `id` (avec ou sans le `get id()` associé) réduit la boucle de feedback, which is a _Good Thing_™️

Reply:
Ca marche, je suis pas forcement convaincu mais je comprends l'interêt 😃

Reply:
> Autre truc que j'aime bien c'est que ça m'amène à faire `myObject.getId()` plutôt que `myObject.id`.
> 
> Dans le premier cas si l'objet n'a pas de fonction `getId()` j'obtiens un `getId is not a function` **en lieu et place de l'appel**. Dans le deuxième cas, si `id` n'existe pas j'obtiens un `undefined` qui, dans le meilleur des cas, provoque un problème 12 mètres plus loin et me laisse le soin de remonter comme un grand à la source du problème et, dans le pire des cas, ne plante pas du tout et potentiellement a des effets néfastes silencieux dont je me rends compte à retardement (Ok, avec les tests qui vont bien ça ne devrait pas se produire mais on est pas infaillibles).
> 
> Donc utiliser `getId()` plutôt que `id` (avec ou sans le `get id()` associé) réduit la boucle de feedback, which is a _Good Thing_

Mais si ta méthode ```getId()``` return un id undefined, ça ne revient pas au même ?

Reply:
@P-Jeremy je parle du fait que tu t'attend à manipuler un CertificationCourse (qui a donc getId() ET un id) et si ça se trouve pour une question X ou Y tu manipules autre chose genre une liste de CertificationCourse... Si effectivement tu manipules un CertificationCourse qui n'a pas de valeur pour son id là oui t'es pas bien dans les deux cas... D'où l'interêt de valider les objets au moment de leur construction.